app/parse.py

```diff
@@ -15,8 +20,106 @@ class Product:
     num_of_reviews: int
 
 
+class WebDriverChrome:
+    def __init__(self, driver: None | WebDriver = None) -> None:
+        self._driver = driver
+
+    @property
+    def driver_chrome(self) -> None | WebDriver:
+        return self._driver
+
+    @driver_chrome.setter
+    def driver_chrome(self, value: WebDriver) -> None:
+        if isinstance(value, WebDriver):
+            self._driver = value
+        else:
+            raise Exception("Driver is invalid")
+
+
+def parse_single_product(product_soup: WebElement) -> Product:
+    count = 0
+    for _ in product_soup.find_elements(By.CSS_SELECTOR, ".glyphicon-star"):
+        count += 1
+    return Product(
+        title=product_soup.find_element(By.CLASS_NAME, "title").text,
+        description=product_soup.find_element(
+            By.CLASS_NAME,
+            "description"
+        ).text,
+        price=float(product_soup.find_element(
+            By.CLASS_NAME,
+            "price"
+        ).text.replace("$", "")),
+        rating=count,
+        num_of_reviews=int(product_soup.find_element(
+            By.CSS_SELECTOR,
+            "div.ratings > .pull-right"
+        ).text.split()[0]),
+    )
+
+
+def get_page_products(path: str, driver: WebDriver) -> list[Product]:
+    url = urljoin(HOME_URL, path)
+
+    driver.get(url)
+
+    try:
+        button = driver.find_element(
+            By.CLASS_NAME,
+            "ecomerce-items-scroll-more"
+        )
+        if button:
+            while button.value_of_css_property("display") != "none":
+                button.click()
+    except Exception:
+        pass
+
+    products = driver.find_elements(By.CLASS_NAME, "thumbnail")
+
+    return [parse_single_product(product) for product in products]
+
+
+def write_to_csv_file(path: str, products: list[Product]) -> None:
```

Comment:
Why do you need a path here?)
Maybe delete urls from `get_all_products` and move `path_file` variable